### PR TITLE
version.cmake: tweak git log invocation

### PR DESCRIPTION
**Pull request by thac0**
If a user has set log.showSignature = true, then build will break
because DNNL_VERSION_HASH define will be overrun by muliple lines about
the signature of the commit.

If a user has customized core.abbrev-- e.g. Linux kernel developers are
encouraged to set it to 12-- then each build could display a commit hash
string of differing lengths. The git default is "auto", so length of the
hash string could change depending on, e.g., the number of commits in
this repository. That's probably undesirable. By explicitly overriding
abbrev on the command line, it ensures all builds going forward will
generate a hash of the same length (namely, the entire commit hash).

With --oneline one and only one line of output ought to appear, and it
will be the commit hash.

Signed-off-by: Joe Konno <joe.konno@intel.com>

### CODE REVIEW

**Comment by igorsafo** [comment]
Hello @thac0 ,
Thanks for your PR.
Will it be enough to use only `--oneline --no-abbrev-commit`? As you pointed in this case only commit hash will be printed. Also I see that `--no-show-signature` is not supported on old Git:
```sh
$ git --version
git version 1.8.3.1
$ git log --no-show-signature --no-abbrev-commit --oneline -1 --format=%H
fatal: unrecognized argument: --no-show-signature
```

Regards,
Igor Safonov

**Comment by thac0** [comment]
> Will it be enough to use only `--oneline --no-abbrev-commit`? As you pointed in this case only commit hash will be printed. Also I see that `--no-show-signature` is not supported on old Git:

Oof. 1.8.3.1 was released 6 years ago. The primary intent was to fix was for the showSignature thing. If you have to support 1.8.3.1, then I'll have to carry this patch locally.

**Comment by igorsafo** [comment]
Currently we don't have any restriction related to Git version.

Have you encountered a real problem using new Git and showSignature property set to True? I couldn't reproduce the issue on my side and I think the option `--format=%H` filters out everything except commit hash so it doesn't whether log.showSignature is set or not.

**Comment by thac0** [comment]
> Currently we don't have any restriction related to Git version.
> 
> Have you encountered a real problem using new Git and showSignature property set to True? I couldn't reproduce the issue on my side and I think the option `--format=%H` filters out everything except commit hash so it doesn't whether log.showSignature is set or not.

tl;dr - Yes, it can break the build if you've signed the commit, or the latest commit was a merge commit (signed by GitHub).

Enable showSignature globally:
```
[log]
        showSignature = true
```

Then, on a signed commit:
```
$ git log -1 --format=%H
gpg: Signature made Thu 02 Sep 2021 02:10:32 PM PDT
gpg:                using EDDSA key 22E8F306C8FA4BAA2A5F36F3A0303B060918941C
gpg: Good signature from "Joe Konno <joe.konno@intel.com>" [ultimate]
gpg:                 aka "Joe Konno <joe.konno@linux.intel.com>" [ultimate]
2e34d03f4a85db4caf5515270174b866c3231a02
```

Using git version 2.33.0

**Comment by thac0** [comment]
I can't account for CI's failure-- surely a change to a define wouldn't impact the tests. If the define were improper, the pre-processor would have exited with a failure (that's what brought me here).

https://dev.azure.com/mkldnn/dnnl/_build/results?buildId=670&view=logs&j=e3b480bf-17df-574e-e953-03e2ce95b133

**Comment by igorsafo** [comment]
@thac0 Thanks for the explanation!
I found a way to force the same behavior and keep it compatible with old git:
`git -c log.showSignature=false log --no-abbrev-commit --oneline -1 --format=%H`
I verified this change with git versions 1.8.3.1 and 2.25.0.
Unfortunately I wasn't able to push that commit into your branch.
Could you please pull it into your branch? Here is the commit which should be applied on top of yours: https://github.com/igorsafo/oneDNN/commit/ebc387f7cbd42817409a161f8d57c027aad24f57

**Comment by thac0** [comment]
> Could you please pull it into your branch? Here is the commit which should be applied on top of yours: [igorsafo@ebc387f](https://github.com/igorsafo/oneDNN/commit/ebc387f7cbd42817409a161f8d57c027aad24f57)

Incorporated it. Thanks @igorsafo !

**Comment by thac0** [comment]
Latest push does the following:
* rebased atop latest `master`
* incorporate @igorsafo 's command line changes to support venerable git versions

**Review by igorsafo — APPROVED** [review]
Thanks, the changes LGTM!

**Review by mgouicem — APPROVED** [review]
